packages/app/background-jobs-common/src/periodic-jobs/AbstractPeriodicJob.ts

```diff
@@ -0,0 +1,301 @@
+import { randomUUID } from 'node:crypto'
+
+import {
+  type CommonLogger,
+  type ErrorReporter,
+  type TransactionObservabilityManager,
+  isError,
+} from '@lokalise/node-core'
+import { resolveGlobalErrorLogObject } from '@lokalise/node-core'
+import type Redis from 'ioredis'
+import { stdSerializers } from 'pino'
+import type { LockOptions } from 'redis-semaphore'
+import { Mutex } from 'redis-semaphore'
+import type { LockLostCallback } from 'redis-semaphore'
+import type { ToadScheduler } from 'toad-scheduler'
+import { AsyncTask, SimpleIntervalJob } from 'toad-scheduler'
+
+const DEFAULT_JOB_INTERVAL = 60000
+const DEFAULT_EXCLUSIVE_LOCK_SUFFIX = 'EXCLUSIVE'
+
+export type BackgroundJobConfiguration = {
+  /**
+   * Job unique name
+   */
+  jobId: string
+  /**
+   * The interval in milliseconds at which the job should run.
+   */
+  intervalInMs?: number
+  /**
+   * Allows to run the job exclusively in a single instance of the application.
+   * The first consumer that acquires the lock will be the only one to run the job until it stops refreshing the lock.
+   */
+  singleConsumerMode?: {
+    exclusiveLockSuffix?: string
+    enabled: boolean
+    /**
+     * By default, the lock TTL is 2 * intervalInMs, to prevent the lock from expiring before the next execution.
+     */
+    lockTimeout?: number
+
+    /**
+     * Lock will be reset to this value after success, so that other node could potentially acquire the lock after it expires, but in order to prevent immediate acquire
+     */
+    lockTimeoutAfterSuccess?: number
+  }
+  /**
+   * If true, the job will log when it starts and finishes.
+   */
+  shouldLogExecution?: boolean
```

Comment:
Shouldn't we always log start and finish?

Reply:
I see and it makes sense, but if I am not wrong it is `false` by default, I would expect to have it `true` but no strong opinion from my side